structuredheadings/plugin.js

```diff
@@ -322,23 +322,33 @@
               );
           var func;
 
+          // handle the collapsed, not in a heading case
+          if (headings === null) {
+            editor.fire("lockSnapshot", { dontUpdate: true });
+            editor.execCommand("matchHeading");
+            editor.fire("unlockSnapshot");
```

Comment:
without fussing about it too much, we just use the old matchHeading code to figure out the correct heading level (and use snapshot manipulation to prevent adding an extra p -> h undo item in the stack)